stdlib/http/server.pyi

```diff
@@ -50,20 +52,33 @@ class BaseHTTPRequestHandler(socketserver.StreamRequestHandler):
     def parse_request(self) -> bool: ...  # Undocumented
 
 class SimpleHTTPRequestHandler(BaseHTTPRequestHandler):
+    server_version: str
     extensions_map: Dict[str, str]
     if sys.version_info >= (3, 7):
         def __init__(
             self,
             request: bytes,
             client_address: Tuple[str, int],
             server: socketserver.BaseServer,
-            directory: Optional[StrPath] = ...,
+            directory: StrPath | None = ...,
```

Comment:
You just changed the syntax, but looking at the code, using a `Path` for directory can fail  in `translate_path()`.

```suggestion
            directory: str | None = ...,
```

Reply:
Yeah I thought the pipe operator is preferred, instead of using `Optional`.
Good eye, you are totally right about the typing!